lib/eco/treelexer/lexer.py

```diff
@@ -34,9 +34,11 @@ class PatternMatcher(object):
 
     def __init__(self):
         self.pos = 0
+        self.exactmatch = True
```

Comment:
What does this attribute mean?

Reply:
This attribute is needed for lookahead calculation. It tells us if a token was matched by something like `[a-z]+`. In that case the match is not exact and token needs a lookahead of 1. On the other hand matching `abc` is exact as there is no longer match possible and thus the lookahead would be 0.

Reply:
Ah. Should it have a default value?

Reply:
Yes by default matches are exact until a match fails (a character was found that doesn't match which means we return what we've found so far and set the lookahead to 1).

Reply:
OK.